api/v1alpha4/vspherecluster_webhook.go

```diff
@@ -51,3 +58,13 @@ func (c *VSphereCluster) ValidateUpdate(old runtime.Object) error {
 func (c *VSphereCluster) ValidateDelete() error {
 	return nil
 }
+
+func defaultVSphereCluterSpec(s *VSphereClusterSpec) {}
+
+func validateVSphereClusterSpec(s VSphereClusterSpec) field.ErrorList {
+	var allErrs field.ErrorList
+	if s.Thumbprint != "" && s.Insecure != nil && *s.Insecure {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "Insecure"), s.Insecure, "cannot be set to true at the same time as .spec.Thumbprint"))
+	}
```

Comment:
This is an FYI, the struct variable `Insecure` is deprecated and is scheduled for removal in v1alpha4. So this will go away once the PR for removing deprecated types gets merged.